lib/hammer_cli_foreman_tasks/helper.rb

```diff
@@ -3,6 +3,9 @@ module Helper
     # render the progress of the task using polling to the task API
     def task_progress(task_or_id)
       task_id = task_or_id.is_a?(Hash) ? task_or_id['id'] : task_or_id
+      if !task_or_id.is_a?(Hash)
```

Comment:
This wouldn't work if `task_or_id` was the actual id, would it? How about we turn this into a case?

```ruby
task_id = case task_or_id
          when Hash
            task_or_id['id']
          when Array
            Hash[*task_or_id.flatten(1)]['task_id']
          else
            task_or_id
          end

```

Reply:
@adamruzicka updated and tested with traces and subscription manifest refresh and both worked correctly.